### PR TITLE
docs: create API reference page, link class/namespace/... references in navigation

### PR DESCRIPTION
**Pull request by mojomex**
## PR Type

- Improvement

## Description

This PR should make it much user-friendlier to navigate through the API reference.

image

## Review Procedure

Run
```shell
python3 -m pip install -U $(curl -fsSL https://raw.githubusercontent.com/autowarefoundation/autoware-github-actions/main/deploy-docs/mkdocs-requirements.txt)
cd src
mkdocs serve
```
and confirm the API reference tab is working as expected.

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR**: Required to pass before the merge.

### CODE REVIEW

**Review by drwnz — APPROVED** [review]
This is great, thank you!
The only issue I see that "API" is rendered "Api" on the API reference page.
I wonder if it is as simple as renaming the file (although that is ugly in it's own way)...
image

File: docs/api_reference.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Nebula is divided into several modules:
+
+- Common, for type definitions and utilities
```
Comment (drwnz) [inline docs/api_reference.md:3]:
```suggestion
- **Common**, for type definitions and utilities
```

File: docs/api_reference.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+- Decoders, for converting raw packets into pointclouds and performing correction and filtering
```
Comment (drwnz) [inline docs/api_reference.md:4]:
```suggestion
- **Decoders**, for converting raw packets into pointclouds and performing correction and filtering
```

File: docs/api_reference.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+- HW Interfaces, for hardware protocols and socket implementations
```
Comment (drwnz) [inline docs/api_reference.md:5]:
```suggestion
- **HW interfaces**, for hardware protocols and socket implementations
```

File: docs/api_reference.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+- ROS Wrappers, for ROS launch, parameter handling, data publishing and diagnostics
```
Comment (drwnz) [inline docs/api_reference.md:6]:
```suggestion
- **ROS wrappers**, for ROS launch, parameter handling, data publishing and diagnostics
```

**Comment by mojomex** [comment]
image

@drwnz Fixed by adding a `# API Reference` markdown heading manually. MkDocs recognizes that and even uses it in the nav menu now.

**Pull request merged**